q2_feature_classifier/_skl.py

```diff
@@ -75,13 +74,13 @@ def _predict_chunk_with_conf(pipeline, separator, confidence, chunk):
     split_classes = [c.split(separator) for c in pipeline.classes_]
     for seq_id, taxon, class_probs in zip(seq_ids, y, prob_pos):
         taxon = taxon.split(separator)
-        classes = zip(deepcopy(split_classes), class_probs)
+        classes = zip(split_classes, class_probs)
         result = []
-        for level in taxon:
-            classes = [cls for cls in classes if cls[0].pop(0) == level]
-            cum_prob = sum(c[1] for c in classes)
+        for rank, level in enumerate(taxon):
+            cum_prob = sum(cls[1] for cls in classes if cls[0][rank] == level)
             if cum_prob < confidence:
                 break
+            classes = zip(split_classes, class_probs)
```

Comment:
Suggest deleting this line and replacing the original definition of classes with a list rather than a zip object.

I'm fairly confident that the result would be neater, less confusing, and more efficient.

I might be wrong about the efficiency, though, and it might be faster to create the zip object at every iteration of this loop than to create the whole list before iterating. You have done performance tests so you might like to test both.

Reply:
The change in efficiency would be negligible here... But I agree, it would be cleaner to create a list instance once in the beginning of the loop.